pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/BranchConditional.java

```diff
@@ -63,19 +88,15 @@ public boolean branchMatches(String toCompare, String actualBranch) {
         } else if (isEmpty(actualBranch) || isEmpty(toCompare)) {
             return false;
         }
-        // Replace the Git directory separator character (always '/')
-        // with the platform specific directory separator before
-        // invoking Ant's platform specific path matching.
-        String safeCompare = toCompare.replace('/', File.separatorChar);
-        String safeName = actualBranch.replace('/', File.separatorChar);
-        return SelectorUtils.matchPath(safeCompare, safeName, false);
+
+        Comparator c = Comparator.get(comparator, Comparator.GLOB);
+        return c.compare(toCompare, actualBranch);
     }
 
     @Extension
     @Symbol("branch")
     public static class DescriptorImpl extends DeclarativeStageConditionalDescriptor<BranchConditional> {
         @Override
-        @Nonnull
```

Comment:
Why remove?

Reply:
Likely my bad with C&P from the `TagConditional` class